### PR TITLE
Update README for BUILD 2018

### PR DESCRIPTION
**Pull request by brandwe**
@jasoncoolmax

### CODE REVIEW

**Comment by brandwe** [comment]
README of one of our 2 third party library samples that signs the user in to iOS and then calls the Microsoft Graph API as a Search.

File: README.md [hunk 1]
```diff
@@ -79,140 +62,125 @@ $ open QuickStart.xcworkspace
...
+                       
+                   }];
 }
-
 ```
-- When your app requests a token by calling `getToken(...)`, ADAL will attempt to return a token without asking the user for credentials.  If ADAL determines that the user needs to sign in to get a token, it will display a login dialog, collect the user's credentials, and return a token upon successful authentication.  If ADAL is unable to return a token for any reason, it will throw an `AdalException`.
-- Notice that the `AuthenticationResult` object contains a `tokenCacheStoreItem` object that can be used to collect information your app may need.  In the QuickStart, `tokenCacheStoreItem` is used to determine if authenitcation has already occurred.
 
 
 ## Step 5: Build and Run the application
```
Comment (jmprieur) [inline README.md]: This is a different formatting than the first 4 paragraph (which were using # 5.)

**Review by jmprieur — APPROVED** [review]
LGTM
A minor formatting issue in Step 5 to be consistent with the rest.

This is an improvement. We can also discuss being consistent with other samples, but this is more work.

File: README.md [hunk 2]
```diff
@@ -40,34 +26,31 @@ The iOS format for a Redirect URI is:
...
-3. Click on **More Services** in the left hand nav, and choose **Azure Active Directory**.
-4. Click on **App registrations** and choose **Add**.
-5. Enter a friendly name for the application, for example 'DirectorySearcher' and select 'Native' as the Application Type. The **Redirect Uri** is a scheme and string combination that Azure AD will use to return token responses.  Enter a value specific to your application based on the information above. Click on **Create** to create the application.
-6. While still in the Azure portal, choose your application, click on **Settings** and choose **Properties**.
-7. Find the Application ID value and copy it to the clipboard.
-8. Configure Permissions for your application - in the Settings menu, choose the 'Required permissions' section, click on **Add**, then **Select an API**, and select 'Microsoft Graph' (this is the Graph API). Then, click on  **Select Permissions** and select 'Read Directory Data'.
+1. Sign in to the [Microsoft Developer Registration Protal](https://apps.dev.microsoft.com/).
```
Comment (danieldobalian) [inline README.md]: Application Registration Portal

File: README.md [hunk 3]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
 ```
 
 -	In the QuickStart project, open the plist file `settings.plist`.  Replace the values of the elements in the section to reflect the values you input into the Azure Portal.  Your code will reference these values whenever it uses ADAL.
-    -	The `tenant` is the domain of your Azure AD tenant, e.g. contoso.onmicrosoft.com
     -	The `clientId` is the clientId of your application you copied from the portal.
     -	The `redirectUri` is the redirect url you registered in the portal.
+    -   The `graphAPI` is the Graph API we will be calling in the QuickStart. Here it is http://graph.microsoft.com
```
Comment (danieldobalian) [inline README.md:59]: https

File: README.md [hunk 4]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
...
+    -   The `api-version` is the API version of the Microsoft Graph API you wish to call.
+    -   The `scopes` are the scopes that you will need to send to the identity service to access the Microsoft Graph API resources. Since this QuickStart will query the organizational directory on your behalf, you'll want Delegated Permissions of `Directory.Read.All` `User.Read.All`, as well as the default OAuth2 scopes `openid` and `offline_access`. Note that if you were using our Microsoft Authentication Library these default scopes would be sent for you. More information about Microsoft Graph API permissions [can be read here](https://developer.microsoft.com/en-us/graph/docs/concepts/permissions_reference).
```
Comment (danieldobalian) [inline README.md:61]: We do not send `Directory.Read.All`, `User.Read.All` in MSAL by default.

File: README.md [hunk 5]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
...
+    
 
-## *4.	Use ADAL to Get Tokens from AAD*
-The basic principle behind ADAL is that whenever your app needs an access token, it simply calls a completionBlock `+(void) getToken : `, and ADAL does the rest.  
-
--	In the `QuickStart` project, open `GraphAPICaller.m` and locate the `// TODO: getToken for generic Web API flows. Returns a token with no additional parameters provided.` comment near the top.  This is where you pass ADAL the coordinates through a CompletionBlock to communicate with Azure AD and tell it how to cache tokens.
+3.	Use Third Party Library to Get Tokens from AAD
```
Comment (danieldobalian) [inline README.md:64]: AAD -> Azure AD

File: README.md [hunk 6]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
...
+We recommend you visit the Third Party LIbrary README for more information on how to use the library to call OAuth2 identity providers. You can find our particular implementation in `LoginViewController.m\h`
```
Comment (danieldobalian) [inline README.md:65]: LIbrary -> Library

Comment (danieldobalian) [reply README.md:65]: Link the library README

File: README.md [hunk 7]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
...
+                                                      NSError *error = [aNotification.userInfo objectForKey:NXOAuth2AccountStoreErrorKey];
+                                                      NSLog(@"Error!! %@", error.localizedDescription);
+                                                  }];
 }
 
+
 ```
 
-- Now we need to use this token to search for users in the graph. Find the `// TODO: implement SearchUsersList` commentThis method makes a GET request to the Azure AD Graph API to query for users whose UPN begins with the given search term.  But in order to query the Graph API, you need to include an access_token in the `Authorization` header of the request - this is where ADAL comes in.
+- Now we need to use this token to search for users in the graph. We provide this in the `graphAPICaller.m/h` as an example. This method makes a GET request to the Azure AD Graph API to query for users whose UPN begins with the given search term.  But in order to query the Graph API, you need to include an access_token in the `Authorization` header of the request - this is where ADAL comes in.
```
Comment (danieldobalian) [inline README.md:102]: Azure AD Graph API? Should this be Microsoft Graph API?

Comment (danieldobalian) [reply README.md:102]: Remove mention of ADAL.

File: README.md [hunk 8]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
...
+                           completionBlock(Users, nil);
+                       }
+                       else
+                       {
+                           completionBlock(nil, error);
+                       }
+                       
+                   }];
+}
+```
```
Comment (danieldobalian) [inline README.md:168]: This is a ton of code to leave in the README. Can it be included in the code and explain what it does in prose? Like, in an "about the code" section?

File: README.md [hunk 9]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
...
-                 }
-                 else
-                 {
-                     completionBlock(nil, error);
-                 }
 
-             }];
-         }
-     }];
+Congratulations! You now have a working iOS application using a third party library that has the ability to authenticate users, securely call Web APIs using OAuth 2.0, and get basic information about the user.  If you haven't already, now is the time to populate your tenant with some users.  Run your QuickStart app, and sign in with one of those users.  Search for other users based on their UPN.  Close the app, and re-run it.  Notice how the user's session remains intact.
```
Comment (danieldobalian) [inline README.md:175]: Earlier this seemed to be a directory searcher app, but this alludes to it just calling the /me endpoint. Can you clarify what the app does?

File: README.md [hunk 10]
```diff
@@ -1,73 +1,48 @@
...
--	Searches a directory for users with a given alias.
-
-To build the complete working application, you’ll need to:
-
-1. Register your application with Azure AD.
-2. Install & Configure ADAL.
-3. Use ADAL to get tokens from Azure AD.
-
-To get started, [download the completed sample](https://github.com/AzureADQuickStarts/NativeClient-iOS/archive/complete.zip).  You'll also need an Azure AD tenant in which you can create users and register an application.  If you don't already have a tenant, [learn how to get one](active-directory-howto-tenant.md).
+# An iOS Microsoft Graph API application with Azure AD v2.0 using third party libraries
```
Comment (danieldobalian) [inline README.md:12]: I recommend introducing the following links: Library source code, AAD Dev landing page, MSAL ObjC, and the v2 library doc.  Also, any reference docs the library exposes.

File: README.md [hunk 11]
```diff
@@ -1,73 +1,48 @@
...
-platforms: ios
+--- 
+Services: active-directory
+platforms: iOS
 author: brandwe
+level: 100
+client: iOS Mobile App
+service: Microsoft Graph
+endpoint: AAD V2
 ---
```
Comment (danieldobalian) [inline README.md:9]: In the description of the repo, update to indicate the sample calls MS Graph. Also, add tags of the sample and what it does.

File: README.md (hunk 4 above)
Comment (danieldobalian) [reply README.md:61]: The code is also requesting `Directory.AccessAsUser.All` and `User.Read`, please add to this description.

File: README.md [hunk 12]
```diff
@@ -1,73 +1,48 @@
...
-An example for this QuickStart code would be: ***msquickstart://com.microsoft.azureactivedirectory.samples.graph.QuickStart***
+1. Register your application
+To enable your app to get tokens, you'll first need to register it in your Azure AD tenant.
 
-## *2. Register the DirectorySearcher Application*
-To enable your app to get tokens, you'll first need to register it in your Azure AD tenant and grant it permission to access the Azure AD Graph API:
+1. Sign in to the [Microsoft Developer Registration Protal](https://apps.dev.microsoft.com/).
+2. Click on **Add An App** in the right corner.
+3. Enter a friendly name for the application, for example 'DirectorySearcher'
+4. Press **Add Platform** and select 'Native' as the Application Type. You will receive an updated set of fields that includes your **Redirect Uri**. The **Redirect Uri** is a scheme and string combination that Azure AD will use to return token responses. Since this is a third party library, you can use the built-in Reply URI of `urn:ietf:wg:oauth:2.0:oob`
```
Comment (danieldobalian) [inline README.md:27]: Use consistent labels of the URI. Mixed Redirect URI and Reply URI.

File: README.md [hunk 13]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
 ```
 
 -	In the QuickStart project, open the plist file `settings.plist`.  Replace the values of the elements in the section to reflect the values you input into the Azure Portal.  Your code will reference these values whenever it uses ADAL.
-    -	The `tenant` is the domain of your Azure AD tenant, e.g. contoso.onmicrosoft.com
     -	The `clientId` is the clientId of your application you copied from the portal.
     -	The `redirectUri` is the redirect url you registered in the portal.
```
Comment (danieldobalian) [inline README.md:58]: I'd clarify they didn't register the Redirect URI, but had it pre-registered.

File: README.md (hunk 3 above)
Comment (danieldobalian) [reply README.md:59]: It's unclear what the URI means and how it's used. Consider adding a description about what `https://graph.microsoft.com` represents.

File: README.md [hunk 14]
```diff
@@ -79,142 +54,165 @@ $ open QuickStart.xcworkspace
...
+    -   The `graphAPI` is the Graph API we will be calling in the QuickStart. Here it is http://graph.microsoft.com
+    -   The `api-version` is the API version of the Microsoft Graph API you wish to call.
```
Comment (danieldobalian) [inline README.md:60]: Consider linking to any description of the various versions and when to use each.

File: README.md (hunk 3 above)
Comment (danieldobalian) [reply README.md:59]: Put the link in "tick marks" so it's clear they're not supposed to click on it.

File: README.md (hunk 11 above)
Comment (danieldobalian) [reply README.md:9]: Continuous integration and build badge.

File: README.md [hunk 15]
```diff
@@ -1,73 +1,48 @@
...
-```
-<app-scheme>://<bundle-id>
-```
-
-- 	**aap-scheme** - This is registered in your XCode project. It is how other applications can call you. You can find this under Info.plist -> URL types -> URL Identifier. You should create one if you don't already have one or more configured.
-- 	**bundle-id** - This is the Bundle Identifier found under "identity" un your project settings in XCode.
+## Steps to Run
 
-An example for this QuickStart code would be: ***msquickstart://com.microsoft.azureactivedirectory.samples.graph.QuickStart***
+1. Register your application
```
Comment (danieldobalian) [inline README.md:21]: This section uses bold, tick marks, and not marking stuff fairly inconsistently.

File: README.md [hunk 16]
```diff
@@ -1,73 +1,48 @@
...
+To enable your app to get tokens, you'll first need to register it in your Azure AD tenant.
 
-## *2. Register the DirectorySearcher Application*
-To enable your app to get tokens, you'll first need to register it in your Azure AD tenant and grant it permission to access the Azure AD Graph API:
+1. Sign in to the [Microsoft Developer Registration Protal](https://apps.dev.microsoft.com/).
+2. Click on **Add An App** in the right corner.
+3. Enter a friendly name for the application, for example 'DirectorySearcher'
```
Comment (danieldobalian) [inline README.md:26]: I think you meant to use ` instead of '.

**Review by danieldobalian — COMMENTED** [review]
Overall, I think the steps are a bit complicated and not necessary. They should be able to download and run the sample (with preconfigured app reg).  If you do want them to copy and paste code, produce a completed version with the code.

**Pull request merged**